modules/gapi/src/api/gproto.cpp

```diff
@@ -132,6 +132,50 @@ cv::GMetaArg cv::descr_of(const cv::GRunArgP &argp)
     }
 }
 
+bool cv::can_describe(const GMetaArg& meta, const GRunArgP& argp)
+{
+    switch (argp.index())
+    {
+#if !defined(GAPI_STANDALONE)
+    case GRunArgP::index_of<cv::Mat*>():               return util::holds_alternative<GMatDesc>(meta) &&
+                                                              util::get<GMatDesc>(meta).canDescribe(*util::get<cv::Mat*>(argp));
+    case GRunArgP::index_of<cv::UMat*>():
+    case GRunArgP::index_of<cv::Scalar*>():            return meta == descr_of(argp);
+#endif //  !defined(GAPI_STANDALONE)
+    case GRunArgP::index_of<cv::gapi::own::Mat*>():    return util::holds_alternative<GMatDesc>(meta) &&
+                                                              util::get<GMatDesc>(meta).canDescribe(*util::get<cv::gapi::own::Mat*>(argp));
+    case GRunArgP::index_of<cv::gapi::own::Scalar*>():
+    case GRunArgP::index_of<cv::detail::VectorRef>():  return meta == descr_of(argp);
+    default: util::throw_error(std::logic_error("Unsupported GRunArgP type"));
+    }
+}
+
+bool cv::can_describe(const GMetaArg& meta, const GRunArg& arg)
+{
+    switch (arg.index())
+    {
+#if !defined(GAPI_STANDALONE)
+    case GRunArg::index_of<cv::Mat>():               return util::holds_alternative<GMatDesc>(meta) &&
+                                                            util::get<GMatDesc>(meta).canDescribe(util::get<cv::Mat>(arg));
+    case GRunArg::index_of<cv::UMat>():
+    case GRunArg::index_of<cv::Scalar>():            return meta == descr_of(arg);
+#endif //  !defined(GAPI_STANDALONE)
+    case GRunArg::index_of<cv::gapi::own::Mat>():    return util::holds_alternative<GMatDesc>(meta) &&
+                                                            util::get<GMatDesc>(meta).canDescribe(util::get<cv::gapi::own::Mat>(arg));
+    case GRunArg::index_of<cv::gapi::own::Scalar>():
+    case GRunArg::index_of<cv::detail::VectorRef>(): return meta == descr_of(arg);
+    default: util::throw_error(std::logic_error("Unsupported GRunArg type"));
+    }
+}
+
+bool cv::can_describe(const GMetaArgs &metas, const GRunArgs &args)
+{
+    return std::equal(metas.begin(), metas.end(), args.begin(),
+                      [](const GMetaArg& meta, const GRunArg& arg) {
+                          return can_describe(meta, arg);
+                      });
+}
```

Comment:
So what is the reason behind these functions?

Reply:
They are used into compiler/backends to check if input/output arguments of the computation match inferred meta. It was a `descr_of(RunArgs&) == metas` comparison before, but now we should use `canDescribe` (or match) for such check for mats